modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java

```diff
@@ -227,6 +234,68 @@ public void postProcessOperationsWithModelsTestWithModelNamePrefix() {
         Assert.assertEquals(tsImports.get(0).get("filename"), "./prefixChild");
     }
 
+    @Test
+    public void testApisGeneration() throws IOException {
+
+        File output = Files.createTempDirectory("typescriptnodeclient_").toFile();
+        output.deleteOnExit();
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("typescript-node")
+                .setInputSpec("src/test/resources/3_0/typescript-node/SampleProject.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        final ClientOptInput clientOptInput = configurator.toClientOptInput();
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(clientOptInput).generate();
+        files.forEach(File::deleteOnExit);
+
+        TestUtils.assertFileContains(Paths.get(output + "/api/basicApi.ts"),
+                "* Sample project");
+        TestUtils.assertFileContains(Paths.get(output + "/api/basicApi.ts"),
+                "export class BasicApi {");
+    }
+
+    @Test
+    public void testModelsGeneration() throws IOException {
+
+        File output = Files.createTempDirectory("typescriptnodeclient_").toFile();
+        output.deleteOnExit();
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("typescript-node")
+                .setInputSpec("src/test/resources/3_0/typescript-node/SampleProject.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        final ClientOptInput clientOptInput = configurator.toClientOptInput();
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(clientOptInput).generate();
+        files.forEach(File::deleteOnExit);
+
+        TestUtils.assertFileContains(Paths.get(output + "/model/group.ts"),
+                "export class Group {");
+    }
+
+    @Test
+    public void testDeprecatedAttribute() throws IOException {
+
+        File output = Files.createTempDirectory("typescriptnodeclient_").toFile();
+        output.deleteOnExit();
+
+        final CodegenConfigurator configurator = new CodegenConfigurator()
+                .setGeneratorName("typescript-node")
+                .setInputSpec("src/test/resources/3_0/typescript-node/SampleProject.yaml")
+                .setOutputDir(output.getAbsolutePath().replace("\\", "/"));
+
+        final ClientOptInput clientOptInput = configurator.toClientOptInput();
+        DefaultGenerator generator = new DefaultGenerator();
+        List<File> files = generator.opts(clientOptInput).generate();
+        files.forEach(File::deleteOnExit);
+
+        TestUtils.assertFileContains(Paths.get(output + "/model/group.ts"),
+                "* @deprecated");
```

Comment:
I see a few issues with this:
* The mustache template content is now bound to this particular unit test
* The match is quite imprecise, if I put the token `@deprecated` anywhere in any description of any variable of this model, the match here would be truthy

I think the sample output below should be sufficient?

Reply:
Indeed not the best approach, but this is similar in most generators where tests search for patterns. The alternative is maybe what it was done with the [Go Server Generator](https://github.com/OpenAPITools/openapi-generator/pull/19504), adding a workflow that tests the generated samples and checks the specific file.
What do you think?

Reply:
`rely on the generated sample output`? without a test? I am not sure I understand.

I think we need a test, either in `TypeScriptNodeClientCodegenTest` or in a workflow verifying the samples (those tests can be run locally, not only by the workflow).

Reply:
@macjohnny can correct me if I am wrong, but our current setup is using the generated samples as a form of snapshot test.
A diff in them will either show up in the pull request or fail on CI.


Think of it like this: 
* You introduce the new logic to add a deprecated marker to generated code
* In the future someone else makes a PR that accidentally changes that logic, so the markers get lost
* Their PR would fail CI
* They would either fix the logic or regenerate the samples 
* If they regenerate the samples it would show as a diff
* A reviewer would decide whether the removal of the marker is intended, or not
* If not, they'll prompt the code to be updated to not remove the markers, possibly with a reference to this pull request

Reply:
The CI doesn't fail if there are changes in the samples as part of the PR. Yes, they are visible to the reviewer (git diff).

The CI fails if the workflow includes tests that verify the samples (that was my previous point). I am not a reviewer, but I'd really want to check via automated tests as much as possible. I have implemented this approach in the Postman and GoServer generators, and I think it is valuable (indeed you had a valid point about working locally).

Reply:
Okay, so let me recap:

You're saying the diff produced in the generated samples would not fail the CI checks if you didn't update them?

If yes, my bad, I was under the impression that any diff on CI would fail the build no matter where it is produced.

If that's not the case, can we add those particular sample files to be included in the check that exists for other samples?

That would be my preferred way, but let's have @macjohnny weigh in on this as he's been doing this a lot longer

Reply:
@joscha @gcatanese often automated tests are implemented as integration tests that run the generated code and make certain assertions, like https://github.com/OpenAPITools/openapi-generator/blob/90bc1000bd390211fe9a364f64b5b973e28da7ed/samples/client/petstore/typescript-angular-v13-provided-in-root/tests/default/src/test/api.spec.ts#L70

in some cases unit tests like https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/fetch/TypeScriptFetchClientCodegenTest.java can make very specific assertions that are either not covered by the generated samples (one needs to balance number of different samples vs covering each variation in configuration) or are hard to catch and thus merit a specific assertion in a unit test.

Reply:
so usually adding a unit test is a good thing

Reply:
TIL. Would it make sense to fail on any diff that is produced by generating from anything under config/ ?

Reply:
> You're saying the diff produced in the generated samples would not fail the CI checks if you didn't update them?

no, the CI will fail if the samples are outdated, i.e. the changed code produces a diff in samples and the PR does not add this diff means the CI fails. so when the diff is added to the PR, the reviewer needs to manually check whether this is fine. this does not mean it would actually compile, or sometimes a reviewer might just overlook a specific aspect (especially if the diffs are large), which is why additional assertions in unit tests can make sense.

Reply:
Got it. I personally think the generated fixtures are sufficient. Unit tests string.contains() assertions seem quite fragile in comparison.